From f490b506f170a8cab91f5858ce1dad19c1a42847 Mon Sep 17 00:00:00 2001
From: Paul Gofman <pgofman@codeweavers.com>
Date: Tue, 9 Apr 2024 20:36:26 -0600
Subject: [PATCH 1490/1562] Revert "ntdll: Add WINE_DISABLE_WRITE_WATCH env var
 to disable write watch support."

This reverts commit 60b7ce66f2d4c96240dd8baf8e0e9110d4ebd14b.

CW-Bug-Id: #23654
---
 dlls/ntdll/unix/virtual.c | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/dlls/ntdll/unix/virtual.c b/dlls/ntdll/unix/virtual.c
index b4d0cec3a21..6c1dc3a8b88 100644
--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -4998,22 +4998,6 @@ static NTSTATUS allocate_virtual_memory( void **ret, SIZE_T *size_ptr, ULONG typ
     SIZE_T size = *size_ptr;
     NTSTATUS status = STATUS_SUCCESS;
 
-    if (type & MEM_WRITE_WATCH)
-    {
-        static int disable = -1;
-
-        if (disable == -1)
-        {
-            const char *env_var;
-
-            if ((disable = (env_var = getenv("WINE_DISABLE_WRITE_WATCH")) && atoi(env_var)))
-                FIXME("Disabling write watch support.\n");
-        }
-
-        if (disable)
-            return STATUS_NOT_SUPPORTED;
-    }
-
     /* Round parameters to a page boundary */
 
     if (is_beyond_limit( 0, size, working_set_limit )) return STATUS_WORKING_SET_LIMIT_RANGE;
-- 
2.45.2

